core/src/main/java/org/veriblock/sdk/rewards/PopRewardCalculator.java

```diff
@@ -144,16 +144,11 @@ public static int getBestPublicationHeight(List<AltPublication> publications) th
     }
 
     public static BigDecimal calculatePopDifficultyForBlock(List<AltChainBlock> blocksInterval) throws SQLException {
-        if(blocksInterval.size() != config.popRewardSettlementInterval + config.popDifficultyAveragingInterval) {
-            throw new IllegalArgumentException("The amount of blocks must be equal to popRewardSettlementInterval + popDifficultyAveragingInterval");
-        }
-
         BigDecimal difficulty = BigDecimal.ZERO;
 
         Collections.sort(blocksInterval);  // make the ascending order for the blocks in the collection, it needs for the correct calculation of the pop score
 
-        for(int i = 0; i < config.popDifficultyAveragingInterval; ++i)
-        {
+        for(int i = 0; i < blocksInterval.size() - config.popRewardSettlementInterval; ++i)  {
```

Comment:
What if blocksInterval.size() < config.popRewardSettlementInterval?

Reply:
blocksInterval.size() is unchanged

Reply:
You should add a check for this case.